### PR TITLE
SSR-2364: proto FieldGroup component

### PR DESCRIPTION
**Pull request by voduchau**

### CODE REVIEW

File: src/field-group/style.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  }
+  kuc-field-group {
+    display: inline-table;
+  }
+  kuc-field-group[hidden] {
+    display: none;
+  }
+  .kuc-field-group__group {
+    min-width: 517px;
+    padding: 0 8px;
```
Comment (nmanhit) [inline src/field-group/style.ts:32]: ```padding: 0px 8px;```

File: src/field-group/style.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    border: 1px solid #e3e7e8;
+    background-color: #f5f5f5;
+  }
+  .kuc-field-group__group h3 {
+    margin: 0;
```
Comment (nmanhit) [inline src/field-group/style.ts:37]:
```margin: 0px;
padding: 0px;```

File: src/field-group/style.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    padding: 0;
+  }
+  .kuc-field-group__group__toggle {
+    display: flex;
+    align-items: center;
+    border-style: none;
+    position: relative;
+    outline: none;
+    margin: 12px 0 12px 8px;
```
Comment (nmanhit) [inline src/field-group/style.ts:46]:
```
margin: 12px 0px 12px 8px;
```

File: src/field-group/style.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    padding: 4px 8px 4px 24px;
+    color: #333333;
+    font-size: 16px;
+    cursor: pointer;
+    border: 1px solid transparent;
+    background-color: inherit;
+    line-height: 1.5;
+  }
+  .kuc-field-group__group__toggle:disabled {
+    color: rgba(0,0,0,.25);
```
Comment (nmanhit) [inline src/field-group/style.ts:57]: add white spcae between the alpha parameter 
```color: rgba(0, 0, 0, 0.25);```

File: src/field-group/style.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  .kuc-field-group__group__toggle__label {
+    display: block;
+    pointer-events: none;
+  }
+  .kuc-field-group__group__toggle svg {
+    position: absolute;
+    left: 8px;
+  }
+  .kuc-field-group__group__body {
+    padding: 0 8px;
```
Comment (nmanhit) [inline src/field-group/style.ts:77]:
```
padding: 0px 8px;
margin-left: 0px;
```

File: src/field-group/index.ts [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    firstUpdated(): void {
+      this._updateContainerWidth();
+    }
+
+    private _updateContainerWidth() {
+      if (!this._bodyEl) return;
+
+      const DEFAULT_WIDTH = 517;
+      const isBodyHidden = this._bodyEl.hasAttribute("hidden");
+      if (isBodyHidden) {
```
[the remark this answers is not in the document]
Comment (voduchau) [reply src/field-group/index.ts:108]: This because I want to calculate body width when it hidden. we cannot calculate body's width when it hidden

**Pull request merged**